src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -35,36 +73,158 @@ case class ErgoStateContext(lastHeaders: Seq[Header], genesisStateDigest: ADDige
 
   override type M = ErgoStateContext
 
-  override def serializer: Serializer[M] = ErgoStateContextSerializer
+  override def serializer: Serializer[M] = ErgoStateContextSerializer(votingSettings)
+
+  //Check that non-zero votes extracted from block header are correct
+  protected def checkVotes(votes: Array[Byte], epochStarts: Boolean): Unit = {
+    if (votes.count(_ != Parameters.SoftFork) > Parameters.ParamVotesCount) throw new Error("Too many votes")
+
+    val prevVotes = mutable.Buffer[Byte]()
+    votes.foreach { v =>
+      if (prevVotes.contains(v)) throw new Error(s"Double vote in ${votes.mkString}")
```

Comment:
Please follow our regular way to perform validations via `ModifierValidator` instead of `Unit` methods that throws exceptions